### PR TITLE
Show graphql error message

### PR DESCRIPTION
**Pull request by StephaneBischoff**
Fix proposal (very simple) for request #1828 

I was not clear from the original ticket what the requirements were. 
- Like the toast message format might not be ideal. Maybe could indicate that it was a HTTP 400 error for example instead of the entire error message. 
- Also, should we continue to say to go and check the logs for futher details?
- I did not test with proxies. Seems like the request made is different in that case. (not using `responseType: "arraybuffer"`

Let me know what you think. thanks

error-message

### CODE REVIEW

**Comment by liyasthomas** [comment]
@StephaneBischoff, to follow the best practices — you can try making seperate branches for each new features and thereby keep the commit history clean.

On this PR, you can rebase to the current main branch and fix the commit history.

**Comment by kaname-png** [comment]
In my opinion it would be nice if the error message was formatted.

File: packages/hoppscotch-app/helpers/strategies/AxiosStrategy.js [hunk 1]
```diff
@@ -52,12 +52,13 @@ const axiosWithoutProxy = async (req, _store) => {
...
       responseType: "arraybuffer",
     })
-
     return res
   } catch (e) {
     if (axios.isCancel(e)) {
       // eslint-disable-next-line no-throw-literal
       throw "cancellation"
+    } else if (e.response.data) {
+      throw new Error(Buffer.from(e.response.data, "base64"))
```
Comment (liyasthomas) [inline packages/hoppscotch-app/helpers/strategies/AxiosStrategy.js:61]:
```suggestion
      throw new Error(Buffer.from(JSON.stringify(e.response.data, null, 2), "base64"))
```

This should beautify the error message.

Comment (StephaneBischoff) [reply packages/hoppscotch-app/helpers/strategies/AxiosStrategy.js:61]: Just saw that there are unit tests. Will add those.

**Comment by AndrewBastin** [comment]
I actually think the response actually should reflect the error JSON itself purely to not mess with the syntax highlighting. Instead we can have an indicator somewhere and a toast or something like that (@liyasthomas thoughts ?)

**Comment by StephaneBischoff** [comment]
> In my opinion it would be nice if the error message was formatted.

Here is new formatting. Let me know what you think? thanks.

image

**Comment by liyasthomas** [comment]
i guess @AndrewBastin is right. We shouldn't append anything to error message and it's better to show the error object as it is. Though, a JSON formatting for readability won't do any harm.

@StephaneBischof, we can remove the prefix: "Error:" and suffix: "Check console for details" strings from error object.

File: packages/hoppscotch-app/components/graphql/RequestOptions.vue [hunk 2]
```diff
@@ -447,7 +447,7 @@ const runQuery = async () => {
       icon: "done",
     })
   } catch (e: any) {
-    response.value = `${e}. ${t("error.check_console_details")}`
+    response.value = `${e}.\n\n${t("error.check_console_details")}`
     nuxt.value.$loading.finish()
 
     $toast.error(`${e} ${t("error.f12_details")}`, {
```
Comment (liyasthomas) [inline packages/hoppscotch-app/components/graphql/RequestOptions.vue:453]:
```suggestion
    response.value = `${e}`
    nuxt.value.$loading.finish()

    $toast.error(
      `${t("error.something_went_wrong")}. ${t("error.check_console_details")}`,
      {
```

File: packages/hoppscotch-app/components/graphql/RequestOptions.vue [hunk 3]
```diff
@@ -447,10 +447,12 @@ const runQuery = async () => {
...
-    response.value = `${e}. ${t("error.check_console_details")}`
+    response.value = `${e}`
     nuxt.value.$loading.finish()
 
-    $toast.error(`${e} ${t("error.f12_details")}`, {
+    $toast.error(
+      `${t("error.something_went_wrong")}. ${t("error.check_console_details")}`,
+      {
       icon: "error_outline",
     })
```
Comment (liyasthomas) [inline packages/hoppscotch-app/components/graphql/RequestOptions.vue:457]:
```suggestion
    $toast.error(
      `${t("error.something_went_wrong")}. ${t("error.check_console_details")}`,
      {
        icon: "error_outline",
      }
    )
```

**Review by liyasthomas — APPROVED** [review]
LGTM.

**Pull request merged**